tagging_accuracy_report.py

```diff
@@ -0,0 +1,118 @@
+        ).get()['Body'].read().decode('utf-8')
+        automated_tags_json = json.loads(automated_tags_file)
+        return automated_tags_json
+
+    def match_tags(self):
+        no_tags = man_tags_not_in_result = man_tags_not_collected = 0
+        for package_name in self.package_topic_json:
+            # print("Package: " + package_name)
+            manual_tags = set(self.package_topic_json[package_name])
+            # print(manual_tags)
+            try:
+                automated_tags = self.get_automated_tags_for_package(
+                    package_name)
+            except ClientError:
+                print("No tags collected for " + package_name)
+                no_tags += 1
+                continue
+            if 'tags' not in automated_tags \
+                or len(automated_tags['tags']) == 0 \
+                    or len(automated_tags['tags']['_sorted']) == 0:
+                    print(
+                        "No tags available for package {}".format(package_name)
+                    )
+                    no_tags += 1
+                    continue
+            # First check if all manual tags are present in result
+            result = set()
+            for tag_data in automated_tags['tags']['result']:
+                tag_name, tag_score = tag_data
+                result.add(tag_name)
+            # print(result)
+            tags_not_found = manual_tags - result
+            if len(tags_not_found) > 0:
+                print(
+                    """{} not found in result for package {} in automated"""
+                    """ tagging""".format(tags_not_found, package_name)
+                )
+                man_tags_not_in_result += 1
+            else:
+                print("Manual tags of {} are in result".format(package_name))
+                continue
+            # if not, check if all manual tags are present in collected tags
+            all_tags = set()
+            for tag_data in automated_tags['tags']['_sorted']:
+                tag_name, tag_score = tag_data
+                all_tags.add(tag_name)
+            tags_not_found = manual_tags - all_tags
+            if len(tags_not_found) > 0:
+                print("{} not found in all tags of package {} in automated """
+                      """tagging""".format(tags_not_found, package_name))
+                man_tags_not_collected += 1
+            else:
+                print("Collected manual tags for {}".format(package_name))
+        return no_tags, man_tags_not_in_result, man_tags_not_collected
+
+
+def main():
+    if len(sys.argv) < 4:
+        print("Usage: ./tagging_accuracy_report manual_tags_bucket """
+              """automated_tags_bucket manual_tags_json""")
+        sys.exit(0)
+    tar = tagging_accuracy_report(sys.argv[1], sys.argv[2])
+    tar.load_manual_tags_json(
+        sys.argv[3])
+    no_tags, not_in_res, not_collected = tar.match_tags()
+    print("{} packages did not have tags available for them.".format(no_tags))
```

Comment:
is this really interesting measure?

Reply:
No, it's for debugging purposes, I shared the list of all packages we did not have tags for with Frido. Maybe if I can dump this in a more suitable way we can set up a pipeline to run tagger for these packages?

Reply:
I think this will be a one time job - just to make sure our tagging is accurate. In production we will run tagging on analysis fully automatically without any human interaction.